### PR TITLE
presonus-universal-control 4.2.0.96206 (new cask)

### PR DESCRIPTION
**Pull request by Shane-CS**
Add new cask Universal Control by PreSonus for controlling PreSonus hardware.

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/u/universal-control.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+cask "universal-control" do
```
Comment (razvanazamfirei) [inline Casks/u/universal-control.rb:1]: I think we should call this `presonus-universal-control`
```suggestion
cask "presonus-universal-control" do
```

File: Casks/u/universal-control.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version "4.2.0.96206"
+  sha256 "dd9f9a0ba91bd2f8b09adb694eedc17bb520e165d9102d38537a8bcd8333acdb"
+
+  url "https://pae-web.presonusmusic.com/downloads/products/dmg/PreSonus_Universal_Control_v#{version.dots_to_underscores}.dmg",
+      verified: "pae-web.presonusmusic.com/"
+  name "Universal Control"
+  desc "Control your PreSonus hardware"
```
Comment (razvanazamfirei) [inline Casks/u/universal-control.rb:8]:
```suggestion
  desc "PreSonus software control interface"
```

File: Casks/u/universal-control.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "https://www.presonus.com/products/Universal-Control"
+
+  livecheck do
+    url "https://legacy.presonus.com/products/Universal-Control/downloads"
+    strategy :page_match
+    regex(/Universal\sControl\sV(\d+(?:\.\d+)*)/i)
+  end
```
Comment (razvanazamfirei) [inline Casks/u/universal-control.rb:15]:
```suggestion
  livecheck do
    url "https://legacy.presonus.com/products/Universal-Control/downloads"
    regex(/Universal\s?Control\sv?(\d+(?:\.\d+)*)/i)
  end
```

File: Casks/u/universal-control.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on macos: ">= :mojave"
+
+  pkg "PreSonus Universal Control.pkg"
+
+  uninstall pkgutil: "presonus.pkg.Universal.Control.daemon"
+  uninstall pkgutil: "presonus.pkg.Universal.Control.presets"
+  uninstall pkgutil: "presonus.pkg.Universal.Control.controlpanel"
+  uninstall pkgutil: "presonus.pkg.Universal.Control.tb_driver"
+  uninstall launchctl: "com.presonus.ucdaemon"
+  uninstall launchctl: "asp.com.presonus.pae-quantum-audio-plug-in"
```
Comment (razvanazamfirei) [inline Casks/u/universal-control.rb:26]:
```suggestion
  uninstall launchctl: [
              "asp.com.presonus.pae-quantum-audio-plug-in",
              "com.presonus.ucdaemon",
            ],
            pkgutil:   [
              "presonus.pkg.Universal.Control.controlpanel",
              "presonus.pkg.Universal.Control.daemon",
              "presonus.pkg.Universal.Control.presets",
              "presonus.pkg.Universal.Control.tb_driver",
            ]
```

File: Casks/u/universal-control.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+end
```
Comment (razvanazamfirei) [inline Casks/u/universal-control.rb:27]: Please try to add a [zap stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap) for this cask.

**Review by razvanazamfirei — DISMISSED** [review]

File: Casks/u/universal-control.rb (hunk 5 above)
Comment (Shane-CS) [reply Casks/u/universal-control.rb:30]: Sure, I'll look into adding one.

Comment (Shane-CS) [reply Casks/u/universal-control.rb]: Added the stanza.

**Review by bevanjkay — APPROVED** [review]

File: Casks/p/presonus-universal-control.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  url "https://pae-web.presonusmusic.com/downloads/products/dmg/PreSonus_Universal_Control_v#{version.dots_to_underscores}.dmg",
+      verified: "pae-web.presonusmusic.com/"
+  name "Universal Control"
+  desc "PreSonus software control interface"
+  homepage "https://www.presonus.com/products/Universal-Control"
+
+  livecheck do
+    url "https://legacy.presonus.com/products/Universal-Control/downloads"
+    regex(/Universal\s?Control\sv?(\d+(?:\.\d+)*)/i)
```
Comment (bevanjkay) [inline Casks/p/presonus-universal-control.rb:13]: The asterisk in the regex should be another `+`

**Review by bevanjkay — APPROVED** [review]

File: Casks/p/presonus-universal-control.rb (hunk 6 above)
Comment (Shane-CS) [reply Casks/p/presonus-universal-control.rb:13]: You're right, totally missed that.

**Review by bevanjkay — APPROVED** [review]
Thank you @Shane-CS

**Pull request merged**